src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+                    // − Parameter : Threshold used for antenna detection
+                    // (applied in register Andet_Control (@610C), see Error!
+                    // Reference source not found.),
+                    // 7 6 5 4 3 2 1 0
+                    // andet_bot andet_up andet_ithl[1:0] andet_ithh[1:0] andet_en
+                    // − Result Length : 1,
+                    // − Result : 0x00 is OK (antenna is detected),
+                    // different from 0x00 is not OK (no antenna is detected). 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
+                default:
+                    break;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Get or set the timeout when PN532 is in virtual card mode
+        /// </summary>
+        public uint VirtualCardTimeout
+        {
+            get { return _virtualCardTimeout * 50; }
+
+            set
+            {
+                // Timeout defines the time-out only in Virtual card configuration (Mode = 0x02).
+                // In Virtual Card mode, this field is mandatory; whereas in the other mode, it is optional.
+                // This parameter indicates the timeout value with a LSB of 50ms.
+                // There is no timeout control if the value is null (Timeout = 0).
+                // The maximum value for the timeout is 12.75 sec (Timeout = 0xFF). 
+                if (value / 50 > 0xFF)
+                    throw new ArgumentException($"{nameof(VirtualCardTimeout)} can't be more than 12750 milliseconds.");
+                _virtualCardTimeout = value / 50;
+                LogInfo.Log($"{nameof(VirtualCardTimeout)} changed: {SetSecurityAccessModule()}", LogLevel.Debug);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the Security Access Module Mode
+        /// </summary>
+        public SecurityAccessModuleMode SecurityAccessModuleMode
+        {
+            get { return _securityAccessModuleMode; }
+            set { LogInfo.Log($"{nameof(SecurityAccessModuleMode)} changed: {SetSecurityAccessModule()}", LogLevel.Debug); }
+        }
+
+
+        private bool SetSecurityAccessModule()
+        {
+            // Pass the SAM, the virtual card timeout and remove IRQ 
+            Span<byte> toSend = stackalloc byte[3] { (byte)_securityAccessModuleMode, (byte)(_virtualCardTimeout), 0x00 };
+            var ret = WriteCommand(CommandSet.SAMConfiguration, toSend);
+            LogInfo.Log($"{nameof(SetSecurityAccessModule)} Write: {ret}", LogLevel.Debug);
+            if (ret < 0) return false;
+            // We don't expect any result, just that the command went well
+            ret = ReadResponse(CommandSet.SAMConfiguration, Span<byte>.Empty);
+            LogInfo.Log($"{nameof(SetSecurityAccessModule)} read: {ret}", LogLevel.Debug);
+            return ret >= 0;
+        }
+
+        private bool CheckVersion()
```

Comment:
as mentioned before: throw or start name with `Is` or something which suggests returning `Yes`/`No` answer